plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -671,4 +674,181 @@ private function normalize_raw_meta_data( array $data ) : array {
+	 * 1. HPOS and posts tables are in sync, or,
+	 * 2. This is a new shop (in this case also create tables).
+	 * and
+	 * 3. All installed WC plugins are compatible.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--for-new-shop]
+	 * : Enable only if this is a new shop, irrespective of whether tables are in sync.
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * [--with-sync]
+	 * : Also enables sync (if it's currently not enabled).
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * ### EXAMPLES
+	 *
+	 *      # Enable HPOS on new shops.
+	 *      wp wc cot enable --for-new-shop
+	 *
+	 * @param array $args Positional arguments passed to the command.
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 *
+	 * @return void
+	 */
+	public function enable( array $args = array(), array $assoc_args = array() ) {
+		$assoc_args = wp_parse_args(
+			$assoc_args,
+			array(
+				'for-new-shop' => false,
+				'with-sync'    => false,
+			)
+		);
+
+		$enable_hpos = true;
+		WP_CLI::log( __( 'Running pre-enable checks...', 'woocommerce' ) );
+
+		$is_new_shop = \WC_Install::is_new_install();
+		if ( $assoc_args['for-new-shop'] && ! $is_new_shop ) {
+			WP_CLI::warning( __( '[Failed] This is not a new shop, but --for-new-shop flag was passed.', 'woocommerce' ) );
+			$enable_hpos = false;
+		}
+
+		/** Feature controller instance @var FeaturesController $feature_controller */
+		$feature_controller = wc_get_container()->get( FeaturesController::class );
+		$plugin_info        = $feature_controller->get_compatible_plugins_for_feature( 'custom_order_tables', true );
+		if ( count( array_merge( $plugin_info['uncertain'], $plugin_info['incompatible'] ) ) > 0 ) {
+			WP_CLI::warning( __( '[Failed] Some installed plugins are incompatible. Please review the plugins by going to WooCommerce > Settings > Advanced > Features and see the Data storage for orders section.', 'woocommerce' ) );
+			$enable_hpos = false;
+		}
+
+		/** DataSynchronizer instance @var DataSynchronizer $data_synchronizer */
+		$data_synchronizer = wc_get_container()->get( DataSynchronizer::class );
+		$pending_orders    = $data_synchronizer->get_total_pending_count();
+		$table_exists      = $data_synchronizer->check_orders_table_exists();
+
+		if ( ! $table_exists ) {
+			WP_CLI::warning( __( 'Orders table does not exist. Creating...', 'woocommerce' ) );
+			if ( $is_new_shop || 0 === $pending_orders ) {
+				$data_synchronizer->create_database_tables();
+				if ( $data_synchronizer->check_orders_table_exists() ) {
+					WP_CLI::log( __( 'Orders table created.', 'woocommerce' ) );
+					$table_exists = true;
+				} else {
+					WP_CLI::warning( __( '[Failed] Orders table could not be created.', 'woocommerce' ) );
+					$enable_hpos = false;
+				}
+			} else {
+				WP_CLI::warning( __( '[Failed] The orders table does not exist and this is not a new shop. Please create the table by going to WooCommerce > Settings > Advanced > Features and enabling sync.', 'woocommerce' ) );
```

Comment:
I am conflicted about this as well, in fact in this CLI command itself, we create the table but only if it's a new shop. It would be a good idea to create the table when sync command runs, I agree. Let me add that.

Reply:
Added here https://github.com/woocommerce/woocommerce/pull/39865/commits/5d858e07f86129df8dec6bf4630c3545489c21f5#diff-13263072ce10ac901c6669d77abeb53267e74303734f1d78a06ea4c9217c929eR167-R176. Agreed that whoever is running sync, also expects table to be present/created.